sample-compose/src/main/java/com/squareup/radiography/sample/compose/ComposeSampleApp.kt

```diff
@@ -131,12 +145,8 @@ internal const val LIVE_HIERARCHY_TEST_TAG = "live-hierarchy"
         Text("SHOW STRING RENDERING DIALOG")
       }
 
-      SideEffect {
-        liveHierarchy.value = Radiography.scan(
-          viewStateRenderers = DefaultsIncludingPii,
-          // Don't trigger infinite recursion.
-          viewFilter = skipComposeTestTagsFilter(LIVE_HIERARCHY_TEST_TAG)
-        )
+      LaunchedEffect(Unit) {
+        runLiveScanning { liveHierarchy.value = it }
```

Comment:
cool so this effectively says "rescan every time we've recomposed". btw does that not retrigger a composition / any risk of infinite loop?

Reply:
Correct. It does not trigger infinite composition loop, although I'm not super clear on why not. I can look into it and document, because it does seem like that should happen.

Reply:
Ok I get why there's no infinite loop now. Because we exclude the scan result composable from the scan, a scan in which only the scan output changes will result in the scan string being set to the same value, which won't invalidate any recompose scopes.